### ISSUE TITLE
Please, consider iso-8601 formatted dates in bare html

### ISSUE THREAD

**Issue by lockywolf** [cppalliance/temp-site#337]

**Comment by alandefreitas** [comment]
> This way the original html will be machine-friendly

In this case, what are the benefits of that?

**Comment by lockywolf** [comment]
>In this case, what are the benefits of that?

I am kinda baffled by this question. Are boost users programmers, or what? 

https://xkcd.com/1179/

I mean, if #336 is implemented, there is not much need to do that, but that issue is a lot of work, whereas changing an existing date is effectively equivalent to selecting a different format string in a call to `.format()`,  which is easy.

**Comment by vinniefalco** [comment]
![6a6d1124cf69e5588588bc7e397598f6](https://github.com/cppalliance/temp-site/assets/1503976/67e7ab22-edff-4cfb-8b8d-064e63124b52)

**Comment by vinniefalco** [comment]
I'm a programmer during work time but when I browse the Internet I typically don't want to have to also write a JavaScript program. That said, iso-8601 formatted dates don't really fit in with the visual motif of the website. In fact the news page is just a placeholder for the page which has not yet been designed. The actual design will show an elapsed time, e.g. "today", "yesterday", "2d", "a week", "a month", "3y" for example. And in low-contrast small text an actual date but it would be formatted client-side according to the user's locale.

**Comment by lockywolf** [comment]
>The actual design will show an elapsed time, e.g. "today", "yesterday", "2d", "a week", "a month", "3y" for example. 

Jesus Chris! Please, no! I have no idea who could have ever came up with such a horrible suggestion, but please, reconsider. It's literally the worst idea I have seen in UX ever.

When I am discussing things with teammates, I want to know whether they could have seen the message or not. "yesterday" makes this hard, and "3y" totally impossible. Moreover, "yesterday" is not even the same yesterday everywhere on Earth, and we have a distributed team.

>I typically don't want to have to also write a JavaScript program in order to do so.

I didn't suggest that. I mean that this is the place where I would suggest adding a small JS scriptlet, which would replace the iso-date with something user-friedlier, which could be tweaked in a user profile. And for potential bots and non-JS users, an iso date would be just a little friendlier.

**Comment by Lastique** [comment]
> The actual design will show an elapsed time, e.g. "today", "yesterday", "2d", "a week", "a month", "3y" for example.

Please don't do this, or at least make this optional. This style of date presentation does not give the reader the knowledge on when the event happened, and the rough idea like "a week ago" isn't really useful. When I'm looking for a date (that is, when I care and need one), I'm most likely interested in a precise time point, not a general description that requires my mental effort to calculate the date and doesn't give the precise date anyway. I really hate this "frivolous" date presentation on other websites.

Regarding this issue, I too think there is value in having the dates in fixed standard machine-friendly format in the sources, so that the dates can be presented to users according to their locales or preferences (if the website is going to implement user-specific preferences at some point).

**Comment by vinniefalco** [comment]
>  there is value in having the dates in fixed standard machine-friendly format in the sources

Yeah that's doable and of course we have the data in the database. We could also show it on the card for the news item in low-contrast, off to the corner of the card. This would be in addition to the elapsed time.
